gcp/src/main/java/org/apache/iceberg/gcp/gcs/GCSFileIO.java

```diff
@@ -209,25 +180,65 @@ private void initMetrics(Map<String, String> props) {
     }
   }
 
+  private Cache<String, PrefixedStorage> storageCache() {
+    if (null == storageCache) {
+      synchronized (this) {
+        if (null == storageCache) {
+          this.storageCache =
+              Caffeine.newBuilder()
```

Comment:
Why not put the cache initialization in the constructor as opposed to here?  We're always going to create the cache even for a single value and the storage clients are lazily initialized via the supplier anyway.

Reply:
I guess we're doing this to make it serializable, right?

Reply:
yes, one reason was to make this serializable and the other reason was to propagate the closure of the storage client, but I guess we can achieve the same by using a simple HashMap